test/testUtilities.ts

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable eslint-comments/disable-enable-pair */
+/* eslint-disable @typescript-eslint/strict-boolean-expressions */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable no-prototype-builtins */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
+import * as vscode from "vscode";
+const SECOND = 1000;
+export const TIMEOUT = 30 * SECOND;
+
+
+export class FakeExtensionContext implements vscode.ExtensionContext {
+    public subscriptions: Array<{ dispose(): any }> = [];
+    public workspaceState: vscode.Memento = new FakeMemento();
+    public globalState: vscode.Memento = new FakeMemento();
+    public storagePath: string | undefined;
+    public globalStoragePath = '.';
+    public logPath = '';
+    public extensionMode;
+    public extensionUri;
+    public environmentVariableCollection;
+
```

Comment:
the context has another three fields as we found yesterday - storageUri, globalStorageUri, logUri, probably need to be added here.

Reply:
Seems like the new fields were added in vscode API 1.49.x. Since this file can go out of sync depending on changes in vscode API version, we should do the following : 

- Rebase this file on new API.
- Add a comment regarding vscode api version being used, so that it is explicit.

Also, how are other extension tests handling this? Are they also maintaining and updating these fake contexts manually on new releases?

Reply:
Will rebase on new API and add comment for new api is introduced in which version.

For other extensions, they use package-lock.json to lock the version of vscode dependency(which is git ignored in our repo). I think both solution has pros and cons. 

Keeping the package-lock.json make all developers  to use same dependency version, which make the development experience smooth, it could also avoid some breaking changes in latest dependency version. However, by not keeping package-lock.json, we could develop with the latest dependency version.

After comparing, I personally vote for keeping package-lock.json. Mike also keep it in redfox repo.

Reply:
Let's keep the `package-lock.json` then. It is better to have a smoother, consistent experience across all devs working on the extension, than having different versions in individual machines leading to different errors/experience changes.

Reply:
Got it, will keep package-lock